modules/context/repo.go

```diff
@@ -202,6 +202,9 @@ func ComposeGoGetImport(owner, repo string) string {
 func EarlyResponseForGoGetMeta(ctx *Context) {
```

Comment:
Same here we can check previously to calling this method and maybe pass `username` and `reponame` as args since they are already extracted and cleaned from context before calling:
https://github.com/go-gitea/gitea/blob/85202d4784758573f80f93dbbda97a444e7ece99/routers/repo/http.go#L58

Reply:
Would it not be preferable to add this for `username` as well in case only `.git` is appended to the Gitea instance URL?

Reply:
I guess not because it seems possible to have the suffix `.git` in a username or organization name.